tests/pytorch/test_pytorch_autolog.py

```diff
@@ -73,9 +87,19 @@ def test_pytorch_autolog_logs_expected_data(pytorch_model):
     _, run = pytorch_model
     data = run.data
 
-    # Checking if metrics are logged
+    # Checking if metrics are logged.
+    # When autolog is configured with the default configuration to not log on steps,
+    # then all metrics are logged per epoch, including step based metrics.
     client = mlflow.tracking.MlflowClient()
-    for metric_key in ["loss", "train_acc", "val_loss", "val_acc"]:
+    for metric_key in [
+        "loss",
+        "train_acc",
+        "val_loss",
+        "val_acc",
+        "loss_forked",
+        "loss_forked_step",
+        "loss_forked_epoch",
```

Comment:
Q. What is the difference between `loss_forked` and `loss_forked_epoch`?

Reply:
They seem identical:

<img width="1630" alt="image" src="https://user-images.githubusercontent.com/17039389/158737086-39ad01a0-fdf4-4b08-9775-29fa4a4fac05.png">

Reply:
Yes they should be exactly the same. PyTorch-Lightning records the same value using the original name (`loss_forked`) as well as the "forked" names (`loss_forked_epoch` and `loss_forked_step`)  in the callback metrics. The non-forked name is used when recording both the step values and epoch values, but I exclude the non-forked metric when logging against the step number as otherwise there would be some values logged against step numbers and some against epoch numbers under the same key so the data wouldn't make any sense.

I didn't also exclude the non-forked name when logging at the epoch end as I didn't want to change the existing behaviour there.

Reply:
> I didn't also exclude the non-forked name when logging at the epoch end as I didn't want to change the existing behaviour there.

Makes sense, thanks for the explanation!